variables.tf

```diff
@@ -41,26 +52,31 @@ variable "database_users" {
 # mongodbatlas_advanced_cluster
 
 variable "cluster_name" {
+  description = "Name of the cluster as it appears in Atlas."
   type    = string
   default = "Cluster0"
 }
 
 variable "provider_name" {
+  description = "Cloud service provider on which the servers are provisioned. Possible values are AWS, GCP, AZUER or TENANT."
   type    = string
   default = "AWS"
 }
 
 variable "backing_provider_name" {
+  description = "Cloud service provider on which you provision the host for a multi-tenant cluster."
   type    = string
   default = "AWS"
 }
 
 variable "region_name" {
+  description = "Physical location of your MongoDB cluster."
```

Comment:
Maybe add a link to the MongoDB documentation: https://www.mongodb.com/docs/atlas/cloud-providers-regions/
Or alternatively, an example